src/static/js/chapter.js

```diff
@@ -0,0 +1,127 @@
+//Data Save can be set to on, so let's check it
+function dataSaverEnabled() {
+  let dataSaver = false;
+  if ('connection' in navigator) {
+    dataSaver = navigator.connection.saveData;
+    if (dataSaver) {
+      console.log('DataSaver is enabled');
+    }
+  }
+
+  return dataSaver;
+}
+
+//iOS causes Google Sheets to create a 6000 by 3700 canvas, which annoyingly isn't supported by iOS!
+//Let's check if we have large Canvas support (annoyingly no API for this!)
+function highResolutionCanvasSupported() {
+
+  let largeCanvasSupported = false;
+
+  try {
+    // Set large sized canvas dimensions and draw test rectangle
+    const cvs = document ? document.createElement('canvas') : null;
+    const ctx = cvs && cvs.getContext ? cvs.getContext('2d') : null;
+    cvs.width = 6000;
+    cvs.height = 3700;
+    ctx.fillRect(5999, 3699, 1, 1);
+    largeCanvasSupported = Boolean(ctx.getImageData(5999, 3699, 1, 1).data[3]) == 1;
+  }
+  catch (e) {
+    largeCanvasSupported = false;
+  }
+
+  if (!largeCanvasSupported) {
+    console.log('High resolution canvas images are not supported');
+  }
+
+  return largeCanvasSupported;
+
+};
+
+//Google Sheets is sometimes blocked by companies/proxies so let's check for that
+//Unfortunately no easy way to check this as cannot make non-CORS request to
+//Another domain and can't check status response when CORS domain :-(
+//Below seems to work for my corporate proxy anyway
+const googleSheetsAllowed = async () => {

```

Comment:
What's the browser support for async/await?

Reply:
[Pretty good](https://caniuse.com/#feat=async-functions) and I tested this in IE11 (which doesn't support it) and it doesn't log any errors, but just ignores it. Which is fine as they will stay with images (which are default).

Reply:
@mikegeyser have you any thoughts on this? Am I right to say that this will just be ignored by incompatible browsers? Or is there a better way to write this?

Reply:
AFAIK it will cause syntax errors and break parsing the file. So any other scripts we put in this file would fail to execute. I'm ok with ES6 but worried that someone someday might put something critical in this file and not realize that it would never work on browsers that don't support newer syntax. Maybe we just rename the file to be data viz specific?

I think we should also remove the discussion comments from this file for that reason. Sorry I think I asked you to add it.

Reply:
OK you've persuaded me. I've removed async/await call instead and therefore left the discussion comments in this file.

Could you give it another once over?